diff -Nur -x '*.orig' -x '*~' twinkle-1.4~beta/configure twinkle-1.4~beta.new/configure
--- twinkle-1.4~beta/configure	2009-01-05 05:49:11.000000000 +1100
+++ twinkle-1.4~beta.new/configure	2009-01-14 13:55:12.000000000 +1100
@@ -6772,23 +6772,23 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        { echo "$as_me:$LINENO: checking for libccrtp1 >= 1.6.0" >&5
-echo $ECHO_N "checking for libccrtp1 >= 1.6.0... $ECHO_C" >&6; }
+        { echo "$as_me:$LINENO: checking for libccrtp1 >= 1.7.0" >&5
+echo $ECHO_N "checking for libccrtp1 >= 1.7.0... $ECHO_C" >&6; }
 
-        if $PKG_CONFIG --exists "libccrtp1 >= 1.6.0" ; then
+        if $PKG_CONFIG --exists "libccrtp1 >= 1.7.0" ; then
             { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
             succeeded=yes
 
             { echo "$as_me:$LINENO: checking CCRTP_CFLAGS" >&5
 echo $ECHO_N "checking CCRTP_CFLAGS... $ECHO_C" >&6; }
-            CCRTP_CFLAGS=`$PKG_CONFIG --cflags "libccrtp1 >= 1.6.0"`
+            CCRTP_CFLAGS=`$PKG_CONFIG --cflags "libccrtp1 >= 1.7.0"`
             { echo "$as_me:$LINENO: result: $CCRTP_CFLAGS" >&5
 echo "${ECHO_T}$CCRTP_CFLAGS" >&6; }
 
             { echo "$as_me:$LINENO: checking CCRTP_LIBS" >&5
 echo $ECHO_N "checking CCRTP_LIBS... $ECHO_C" >&6; }
-            CCRTP_LIBS=`$PKG_CONFIG --libs "libccrtp1 >= 1.6.0"`
+            CCRTP_LIBS=`$PKG_CONFIG --libs "libccrtp1 >= 1.7.0"`
             { echo "$as_me:$LINENO: result: $CCRTP_LIBS" >&5
 echo "${ECHO_T}$CCRTP_LIBS" >&6; }
         else
@@ -6796,7 +6796,7 @@
             CCRTP_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            CCRTP_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libccrtp1 >= 1.6.0"`
+            CCRTP_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libccrtp1 >= 1.7.0"`
             echo $CCRTP_PKG_ERRORS
         fi
 
@@ -6811,8 +6811,8 @@
   if test $succeeded = yes; then
      :
   else
-     { { echo "$as_me:$LINENO: error: Library requirements (libccrtp1 >= 1.6.0) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." >&5
-echo "$as_me: error: Library requirements (libccrtp1 >= 1.6.0) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." >&2;}
+     { { echo "$as_me:$LINENO: error: Library requirements (libccrtp1 >= 1.7.0) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." >&5
+echo "$as_me: error: Library requirements (libccrtp1 >= 1.7.0) not met; consider adjusting the PKG_CONFIG_PATH environment variable if your libraries are in a nonstandard prefix so pkg-config can find them." >&2;}
    { (exit 1); exit 1; }; }
   fi
 
diff -Nur -x '*.orig' -x '*~' twinkle-1.4~beta/configure.in twinkle-1.4~beta.new/configure.in
--- twinkle-1.4~beta/configure.in	2009-01-14 13:54:02.000000000 +1100
+++ twinkle-1.4~beta.new/configure.in	2009-01-14 13:54:23.000000000 +1100
@@ -66,7 +66,7 @@
 
 export PKG_CONFIG_PATH
 
-PKG_CHECK_MODULES(CCRTP, libccrtp1 >= 1.6.0)
+PKG_CHECK_MODULES(CCRTP, libccrtp1 >= 1.7.0)
 
 PKG_CHECK_MODULES(XML2, libxml-2.0)
 # AC_CHECK_HEADER(libxml/tree.h, [],
diff -Nur -x '*.orig' -x '*~' twinkle-1.4~beta/README twinkle-1.4~beta.new/README
--- twinkle-1.4~beta/README	2006-10-02 01:27:12.000000000 +1000
+++ twinkle-1.4~beta.new/README	2009-01-14 13:55:30.000000000 +1100
@@ -15,7 +15,7 @@
 libccgnu2 (version >= 1.4.2) [GNU Common C++]
 	http://www.gnu.org/software/commoncpp/
 
-libccrtp1 (version >= 1.5.0) [GNU RTP Stack]
+libccrtp1 (version >= 1.7.0) [GNU RTP Stack]
 libzrtpcpp (version >= 0.9.0) [Extension library of GNU ccRTP]
 	http://www.gnu.org/software/ccrtp/
 
diff -Nur -x '*.orig' -x '*~' twinkle-1.4~beta/twinkle.spec.in twinkle-1.4~beta.new/twinkle.spec.in
--- twinkle-1.4~beta/twinkle.spec.in	2008-08-23 22:16:10.000000000 +1000
+++ twinkle-1.4~beta.new/twinkle.spec.in	2009-01-14 13:56:32.000000000 +1100
@@ -12,7 +12,7 @@
 URL:		http://www.twinklephone.com
 Requires:	alsa
 Requires:	commoncpp2 >= 1.6.0 
-Requires:	ccrtp >= 1.6.0
+Requires:	ccrtp >= 1.7.0
 Requires:	libzrtpcpp >= 1.3.0
 Requires:	kdelibs3 >= 3.2.0
 Requires:	libsndfile
