diff -Nur twinkle-1.0~beta/src/gui/gui.h twinkle-1.0~beta.new/src/gui/gui.h
--- twinkle-1.0~beta/src/gui/gui.h	2006-12-28 16:22:32.000000000 +0000
+++ twinkle-1.0~beta.new/src/gui/gui.h	2007-01-08 20:44:44.000000000 +0000
@@ -214,7 +214,7 @@
 	void cb_retrieve_referrer(t_user *user_config, int line);
 	
 	// A consulation call for a call transfer is being setup.
-	void t_gui::cb_consultation_call_setup(t_user *user_config, int line);
+	void cb_consultation_call_setup(t_user *user_config, int line);
 	
 	// STUN errors
 	void cb_stun_failed(t_user *user_config, int err_code, const string &err_reason);
diff -Nur twinkle-1.0~beta/src/line.h twinkle-1.0~beta.new/src/line.h
--- twinkle-1.0~beta/src/line.h	2006-12-27 09:33:52.000000000 +0000
+++ twinkle-1.0~beta.new/src/line.h	2007-01-08 20:44:30.000000000 +0000
@@ -254,7 +254,7 @@
 	// permission. This response is received on the dialog that received
 	// the REFER before.
 	// The request (r) is the REFER request that was received.
-	void t_line::recvd_refer_permission(bool permission, t_request *r);
+	void recvd_refer_permission(bool permission, t_request *r);
 	
 	void recvd_stun_resp(StunMessage *r, t_tuid tuid, t_tid tid);
 
