--- gui.cpp.orig	2006-02-21 18:28:06.000000000 +0000
+++ src/gui/gui.cpp	2006-02-21 18:28:16.000000000 +0000
@@ -987,7 +987,7 @@
 	s = user_config->get_profile_name().c_str();
 	const list<t_contact_param> &l = r->hdr_contact.contact_list;
 	if (l.size() == 0) {
-		s += ": you are nor registered";
+		s += ": you are not registered";
 		mainWindow->display(s);
 	} else {
 		s += ": you have the following registrations";
