diff -Nur twinkle-0.9/configure.in twinkle-0.9.new/configure.in
--- twinkle-0.9/configure.in	2006-11-25 08:39:02.000000000 +0000
+++ twinkle-0.9.new/configure.in	2006-11-25 08:39:03.000000000 +0000
@@ -140,6 +140,17 @@
 AC_CHECK_LIB(sndfile, sf_open, [], 
 	[AC_MSG_ERROR([libsndfile library is missing.])])
 
+# Check if libgsm is available
+AC_CHECK_LIB(gsm, sf_open, [
+	AC_CHECK_HEADER(gsm.h, [],
+		[AC_MSG_ERROR([gsm header files missing (gsm.h)])])
+	AC_DEFINE(HAVE_GSM, 1, [Define to 1 if you have the <libgsm> library.])
+	LIBS="-lgsm $LIBS"
+	echo "LIBS += -lgsm" >> $QT_INCL_PRO
+	have_gsm="yes" ], [
+	have_gsm="no"
+	echo "$(top_builddir)/src/audio/gsm/libgsm.a" >> $QT_INCL_PRO ])
+
 # Check if ALSA is available
 AC_CHECK_LIB(asound, main, [
 	AC_CHECK_HEADER(alsa/asoundlib.h, [],
@@ -228,3 +239,4 @@
 AC_MSG_RESULT([Speex:       $have_speex])
 AC_MSG_RESULT([iLBC:        $have_ilbc])
 AC_MSG_RESULT([ZRTP:        $have_zrtp])
+AC_MSG_RESULT([Libgsm dynamic link:	$have_gsm])
diff -Nur twinkle-0.9/src/Makefile.am twinkle-0.9.new/src/Makefile.am
--- twinkle-0.9/src/Makefile.am	2006-11-25 08:39:02.000000000 +0000
+++ twinkle-0.9.new/src/Makefile.am	2006-11-25 08:39:03.000000000 +0000
@@ -16,10 +16,10 @@
         $(top_builddir)/src/sockets/libsocket.a\
         $(top_builddir)/src/threads/libthread.a\
         $(top_builddir)/src/audio/libaudio.a\
-        $(top_builddir)/src/audio/gsm/libgsm.a\
         $(top_builddir)/src/audits/libaudits.a\
         $(top_builddir)/src/stun/libstun.a\
 	$(CCRTP_LIBS)\
+	$(LIBS)\
 	-lresolv\
 	-lsndfile\
 	-lboost_regex
diff -Nur twinkle-0.9/src/audio/audio_decoder.h twinkle-0.9.new/src/audio/audio_decoder.h
--- twinkle-0.9/src/audio/audio_decoder.h	2006-11-25 08:39:02.000000000 +0000
+++ twinkle-0.9.new/src/audio/audio_decoder.h	2006-11-25 08:39:03.000000000 +0000
@@ -25,7 +25,12 @@
 #include "twinkle_config.h"
 #include "audio_codecs.h"
 #include "user.h"
+
+#ifdef HAVE_GSM
+#include <gsm/gsm.h>
+#else
 #include "gsm/inc/gsm.h"
+#endif
 
 #ifdef HAVE_SPEEX
 #include <speex/speex.h>
diff -Nur twinkle-0.9/src/audio/audio_encoder.h twinkle-0.9.new/src/audio/audio_encoder.h
--- twinkle-0.9/src/audio/audio_encoder.h	2006-11-25 08:39:02.000000000 +0000
+++ twinkle-0.9.new/src/audio/audio_encoder.h	2006-11-25 08:39:03.000000000 +0000
@@ -25,7 +25,12 @@
 #include "twinkle_config.h"
 #include "audio_codecs.h"
 #include "user.h"
+
+#ifdef HAVE_GSM
+#include <gsm/gsm.h>
+#else
 #include "gsm/inc/gsm.h"
+#endif
 
 #ifdef HAVE_SPEEX
 #include <speex/speex.h>
diff -Nur twinkle-0.9/src/gui/twinkle.pro twinkle-0.9.new/src/gui/twinkle.pro
--- twinkle-0.9/src/gui/twinkle.pro	2006-11-25 08:39:02.000000000 +0000
+++ twinkle-0.9.new/src/gui/twinkle.pro	2006-11-25 08:39:03.000000000 +0000
@@ -3,7 +3,7 @@
 
 CONFIG	+= qt warn_off release thread
 
-LIBS	+= ../libtwinkle.a ../parser/libsipparser.a ../sdp/libsdpparser.a ../sockets/libsocket.a ../threads/libthread.a ../audio/libaudio.a ../audio/gsm/libgsm.a ../audits/libaudits.a ../stun/libstun.a -lresolv -lsndfile -lboost_regex
+LIBS	+= ../libtwinkle.a ../parser/libsipparser.a ../sdp/libsdpparser.a ../sockets/libsocket.a ../threads/libthread.a ../audio/libaudio.a ../audits/libaudits.a ../stun/libstun.a -lresolv -lsndfile -lboost_regex
 
 DEFINES	+= QT_NO_STL
 
diff -Nur twinkle-0.9/src/twinkle_config.h.in twinkle-0.9.new/src/twinkle_config.h.in
--- twinkle-0.9/src/twinkle_config.h.in	2006-10-01 20:11:58.000000000 +0100
+++ twinkle-0.9.new/src/twinkle_config.h.in	2006-11-25 08:39:50.000000000 +0000
@@ -12,6 +12,9 @@
 /* Defines if your system has the crypt function */
 #undef HAVE_CRYPT
 
+/* Define to 1 if you have the <libgsm> library. */
+#undef HAVE_GSM
+
 /* Define to 1 if you have the <ilbc> library. */
 #undef HAVE_ILBC
 
@@ -135,22 +138,22 @@
 /* Define to the version of this package. */
 #undef PACKAGE_VERSION
 
-/* The size of a `char *', as computed by sizeof. */
+/* The size of `char *', as computed by sizeof. */
 #undef SIZEOF_CHAR_P
 
-/* The size of a `int', as computed by sizeof. */
+/* The size of `int', as computed by sizeof. */
 #undef SIZEOF_INT
 
-/* The size of a `long', as computed by sizeof. */
+/* The size of `long', as computed by sizeof. */
 #undef SIZEOF_LONG
 
-/* The size of a `short', as computed by sizeof. */
+/* The size of `short', as computed by sizeof. */
 #undef SIZEOF_SHORT
 
-/* The size of a `size_t', as computed by sizeof. */
+/* The size of `size_t', as computed by sizeof. */
 #undef SIZEOF_SIZE_T
 
-/* The size of a `unsigned long', as computed by sizeof. */
+/* The size of `unsigned long', as computed by sizeof. */
 #undef SIZEOF_UNSIGNED_LONG
 
 /* Define to 1 if you have the ANSI C header files. */
