only in patch2:
unchanged:
--- twinkle-0.2.1.orig/src/audio/audio_session.cpp
+++ twinkle-0.2.1/src/audio/audio_session.cpp
@@ -672,11 +672,13 @@
 
 	if (audio_rx) {
 		try {
+			audio_rx->set_running(true);
 			thr_audio_rx = new t_thread(main_audio_rx, NULL);
 			MEMMAN_NEW(thr_audio_rx);
 			// thr_audio_rx->set_sched_fifo(90);
 			thr_audio_rx->detach();
 		} catch (int) {
+			audio_rx->set_running(false);
 			string msg("Failed to create audio_rx thread.");
 			log_file->write_report(msg, "t_audio_session::run",
 				LOG_NORMAL, LOG_CRITICAL);
@@ -688,11 +690,13 @@
 
 	if (audio_tx) {
 		try {
+			audio_tx->set_running(true);
 			thr_audio_tx = new t_thread(main_audio_tx, NULL);
 			MEMMAN_NEW(thr_audio_tx);
 			// thr_audio_tx->set_sched_fifo(90);
 			thr_audio_tx->detach();
 		} catch (int) {
+			audio_tx->set_running(false);
 			string msg("Failed to create audio_tx thread.");
 			log_file->write_report(msg, "t_audio_session::run",
 				LOG_NORMAL, LOG_CRITICAL);
only in patch2:
unchanged:
--- twinkle-0.2.1.orig/src/audio/audio_rx.cpp
+++ twinkle-0.2.1/src/audio/audio_rx.cpp
@@ -397,6 +397,10 @@
 	}
 }
 
+void t_audio_rx::set_running(bool _is_running) {
+	is_running = _is_running;
+}
+
 void t_audio_rx::run(void) {
 	int status;
 	audio_buf_info dsp_info;
only in patch2:
unchanged:
--- twinkle-0.2.1.orig/src/audio/audio_rx.h
+++ twinkle-0.2.1/src/audio/audio_rx.h
@@ -156,6 +156,8 @@
 
 	~t_audio_rx();
 
+	void set_running(bool is_running);
+
 	void run(void);
 
 	// Set the dynamic payload type for telephone events
only in patch2:
unchanged:
--- twinkle-0.2.1.orig/src/audio/audio_tx.h
+++ twinkle-0.2.1/src/audio/audio_tx.h
@@ -148,6 +148,8 @@
 
 	~t_audio_tx();
 
+	void set_running(bool is_running);
+
 	void run(void);
 	
 	// Set the dynamic payload type for telephone events
only in patch2:
unchanged:
--- twinkle-0.2.1.orig/src/audio/audio_tx.cpp
+++ twinkle-0.2.1/src/audio/audio_tx.cpp
@@ -330,6 +330,10 @@
 	}
 }
 
+void t_audio_tx::set_running(bool _is_running) {
+	is_running = _is_running;
+}
+
 void t_audio_tx::run(void) {
 	const AppDataUnit* adu;
 	struct timespec sleeptimer;
