Index: twinkle-1.4.2/acinclude.m4
===================================================================
--- twinkle-1.4.2.orig/acinclude.m4	2013-07-25 12:01:11.000000000 -0400
+++ twinkle-1.4.2/acinclude.m4	2013-07-25 12:03:05.000000000 -0400
@@ -829,32 +829,7 @@
 #endif
 ])
 
-AC_CHECK_FUNCS([vsnprintf snprintf])
-
-AH_VERBATIM(_TRU64,[
-/*
- * On HP-UX, the declaration of vsnprintf() is needed every time !
- */
-
-#if !defined(HAVE_VSNPRINTF) || defined(hpux)
-#if __STDC__
-#include <stdarg.h>
-#include <stdlib.h>
-#else
-#include <varargs.h>
-#endif
-#ifdef __cplusplus
-extern "C"
-#endif
-int vsnprintf(char *str, size_t n, char const *fmt, va_list ap);
-#ifdef __cplusplus
-extern "C"
-#endif
-int snprintf(char *str, size_t n, char const *fmt, ...);
-#endif
-])
-
-])
+AC_CHECK_FUNCS([snprintf])
 
 dnl ------------------------------------------------------------------------
 dnl Find the header files and libraries for X-Windows. Extended the
@@ -2156,7 +2131,7 @@
   kde_cv_proto_$1=no
 else
   case "$1" in
-	setenv|unsetenv|usleep|random|srandom|seteuid|mkstemps|mkstemp|revoke|vsnprintf|strlcpy|strlcat)
+	setenv|unsetenv|usleep|random|srandom|seteuid|mkstemps|mkstemp|revoke|strlcpy|strlcat)
 		kde_cv_proto_$1="yes - in libkdefakes"
 		;;
 	*)
Index: twinkle-1.4.2/src/twinkle_config.h.in
===================================================================
--- twinkle-1.4.2.orig/src/twinkle_config.h.in	2013-07-25 12:03:36.000000000 -0400
+++ twinkle-1.4.2/src/twinkle_config.h.in	2013-07-25 12:05:13.019259521 -0400
@@ -91,9 +91,6 @@
 /* Define if you have a STL implementation by SGI */
 #undef HAVE_SGI_STL
 
-/* Define to 1 if you have the `snprintf' function. */
-#undef HAVE_SNPRINTF
-
 /* Define to 1 if you have the <speex> library. */
 #undef HAVE_SPEEX
 
@@ -142,9 +139,6 @@
 /* Define to 1 if you have the <unistd.h> header file. */
 #undef HAVE_UNISTD_H
 
-/* Define to 1 if you have the `vsnprintf' function. */
-#undef HAVE_VSNPRINTF
-
 /* Define to 1 if you have the <libzrtpcpp> library. */
 #undef HAVE_ZRTP
 
@@ -275,30 +269,6 @@
 #endif
 
 
-
-/*
- * On HP-UX, the declaration of vsnprintf() is needed every time !
- */
-
-#if !defined(HAVE_VSNPRINTF) || defined(hpux)
-#if __STDC__
-#include <stdarg.h>
-#include <stdlib.h>
-#else
-#include <varargs.h>
-#endif
-#ifdef __cplusplus
-extern "C"
-#endif
-int vsnprintf(char *str, size_t n, char const *fmt, va_list ap);
-#ifdef __cplusplus
-extern "C"
-#endif
-int snprintf(char *str, size_t n, char const *fmt, ...);
-#endif
-
-
-
 #if defined(__SVR4) && !defined(__svr4__)
 #define __svr4__ 1
 #endif
